### DISCUSSION TITLE
Restarting the simulator for each cocotb test factory test

### DISCUSSION THREAD

**Discussion by HobsonPelican** [cocotb/cocotb#3907] [Q&A] [answered]
Hi,

I am using cocotb testfactory. Calling the module results in the generated tests being executed, however the simulator doesn't restart upon each test. Is there a way to restart the simulator on each test ?

**Comment by ktbarrett** [comment, marked as answer]
There is not currently any way to do that.

**Reply by HobsonPelican** [reply]
Thanks @ktbarrett

**Reply by ktbarrett** [reply]
There are homegrown frameworks to do exactly that, but most exist behind closed doors. It's going to take someone with enough motivation (or permission) to make a FOSS version of such a tool. It's not terribly difficult to accomplish:

1. Run a simulation with a test designed to report all tests in a given MODULE.
2. Use that report to spin up several tests in different simulator calls (use [this documentation](https://docs.cocotb.org/en/stable/custom_flows.html) to help with that).
